src/System.IO.FileSystem/src/System/IO/Win32FileStream.cs

```diff
@@ -547,6 +562,16 @@ private void FlushWrite(bool calledFromFinalizer)
                 {
                     writeTask.GetAwaiter().GetResult();
                 }
+
+                // If not completing synchronously, ensure the async flush operation
+                // is tracked in case of incoming WriteAsync calls.
+                else
+                {
+                    if (ActiveBufferOperation())
+                        _activeBufferOperation = Task.WhenAll(_activeBufferOperation, writeTask);
+                    else
+                        _activeBufferOperation = writeTask;
```

Comment:
This is just personal preference, so feel free to leave it as-is, but I prefer to write such things as:
```C#
_activeBufferOperation = ActiveBufferOperation() ?
    Task.WhenAll(writeTask, _activeBufferOperation) :
    _writeTask;
```
That way it's clear that we're always assigning into the field, and the conditional just determines exactly what we're assigning.